app/vmagent/remotewrite/remotewrite.go

```diff
@@ -759,16 +759,29 @@ func (rwctx *remoteWriteCtx) reinitStreamAggr() {
 		logger.Errorf("cannot reload stream aggregation config from -remoteWrite.streamAggr.config=%q; continue using the previously loaded config; error: %s", sasFile, err)
 		return
 	}
-	if !sasNew.Equal(sas) {
-		sasOld := rwctx.sas.Swap(sasNew)
-		sasOld.MustStop()
-		logger.Infof("successfully reloaded stream aggregation configs at -remoteWrite.streamAggr.config=%q", sasFile)
-	} else {
+
+	defer func() {
+		metrics.GetOrCreateCounter(fmt.Sprintf(`vmagent_streamaggr_config_reload_successful{path=%q}`, sasFile)).Set(1)
+		metrics.GetOrCreateCounter(fmt.Sprintf(`vmagent_streamaggr_config_reload_success_timestamp_seconds{path=%q}`, sasFile)).Set(fasttime.UnixTimestamp())
+	}()
+
+	if sasNew.Equal(sas) {
 		sasNew.MustStop()
 		logger.Infof("the config at -remoteWrite.streamAggr.config=%q wasn't changed", sasFile)
+		return
 	}
-	metrics.GetOrCreateCounter(fmt.Sprintf(`vmagent_streamaggr_config_reload_successful{path=%q}`, sasFile)).Set(1)
-	metrics.GetOrCreateCounter(fmt.Sprintf(`vmagent_streamaggr_config_reload_success_timestamp_seconds{path=%q}`, sasFile)).Set(fasttime.UnixTimestamp())
+
+	if sas == nil {
+		sas = &streamaggr.Aggregators{}
+	}
```

Comment:
Seems unneeded, should be covered by https://github.com/VictoriaMetrics/VictoriaMetrics/blob/cc0cc49234df3dc5070a83d40092b2709a86cf07/app/vmagent/remotewrite/remotewrite.go#L746-L749

Since reload can't change `sas` from nil to not-nil.